### DISCUSSION TITLE
Stable version?

### DISCUSSION THREAD

**Discussion by patrickjane** [helmfile/helmfile#1408] [Questions]
I am comparing features between helmfile & helmsman currently, and helmsman is lacking some of the features I need, which helmfile provides.
However, helmsman is version 3.x.x, while helmfile is version 0.x.x and states: _"it is still in its early stage of development, hence versioned 0.x."_.

I am a bit suprised about this, since both projects are roughly of the same age (first releases dating back to 2016-2017), and both projects seem to have roughly the same number of active contributors. I would not consider an 8 year-old project to be "in its early stage of development", to be honest.

When can a stable release be expected for helmfile? I am a bit hesitant regarding the usage of non-stable products in operative environments, especially if breaking API changes will most likely happen along the road.

**Comment by yxxhero** [comment]
see: https://github.com/helmfile/helmfile/blob/main/docs/proposals/towards-1.0.md

**Comment by patrickjane** [comment]
So I could consider helmfile to be stable, and just avoid the announced 3 API breaking changes and be fine?

**Reply by mumoshu** [reply]
Hey!

In short, you can expect both v0.x and unreleased v1.x to be stable, in the sense that v0.x has no planned backward-incompatible changes made within v0.x, and v1.x is supposed to be backward-compatible for the entire v1.x release line.
We take backward compatibility seriously anyway.

For any new users, I suggest starting to use helmfile with the `HELMFILE_V1MODE` envvar as suggested by @yxxhero. That way, upgrading to v1 would be a breeze.

**Comment by yxxhero** [comment]
you can set env `HELMFILE_V1MODE` environment to use v1 mode.
